pkg/controller/backup_storage_location_controller.go

```diff
@@ -77,14 +77,14 @@ func (r *BackupStorageLocationReconciler) Reconcile(req ctrl.Request) (ctrl.Resu
 			defaultFound = true
 		}
 
-		backupStore, err := r.NewBackupStore(location, pluginManager, log)
-		if err != nil {
-			log.WithError(err).Error("Error getting a backup store")
+		if !storage.IsReadyToValidate(location.Spec.ValidationFrequency, location.Status.LastValidationTime, r.DefaultBackupLocationInfo, log) {
```

Comment:
I think this only addresses the case where validation frequency is 0, unless I'm misreading something.

If validation frequency is 1 minute (the server's default), then it will still invoke the Init() function. Is that acceptable in your case?

Reply:
This change ensures we first check if the BSL is ready to be validated and then initialize the backup store only for those BSLs that pass the check. It should address all cases where validation frequency is 0 as well as user specified validation frequency or server's default validation frequency.